pkg/app/piped/cloudprovider/kubernetes/diff.go

```diff
@@ -86,6 +94,48 @@ func DiffList(olds, news []Manifest, opts ...diff.Option) (*DiffListResult, erro
 	return cr, nil
 }
 
+// Following `remarshal` function and comment is copied from argocd/gitops-engine and modified
+// https://github.com/argoproj/gitops-engine/blob/b0c5e00ccfa5d1e73087a18dc59e2e4c72f5f175/pkg/diff/diff.go#L685-L723
+
+// remarshal checks resource kind and version and re-marshal using corresponding struct custom marshaller.
+// This ensures that expected resource state is formatter same as actual resource state in kubernetes
+// and allows to find differences between actual and target states more accurately.
+// Remarshalling also strips any type information (e.g. float64 vs. int) from the unstructured
+// object. This is important for diffing since it will cause godiff to report a false difference.
+func remarshal(obj *unstructured.Unstructured) *unstructured.Unstructured {
+	data, err := json.Marshal(obj)
+	if err != nil {
+		panic(err)
+	}
+	gvk := obj.GroupVersionKind()
+	item, err := scheme.Scheme.New(obj.GroupVersionKind())
+	if err != nil {
+		// This is common. the scheme is not registered
+		log.Printf("Could not create new object of type %s: %v", gvk, err)
```

Comment:
Don't use this `log` package.
Return the error and use our logger to log.

Reply:
This error is common. So can I ignore this error and simply return input object?

Reply:
@Hosshii I mean we can just return the error and add the log here https://github.com/pipe-cd/pipecd/blob/0dd697efa8ef7be0ad3ddf2815e813a721ccf148/pkg/app/piped/cloudprovider/kubernetes/diff.go#L62

Reply:
Does this mean that when there is an error, it returns both the error and the object?
I mean that if error is occurred, return original object and error, then caller side print the error and continue process.

Reply:
When there is an error, return only the error.
And at the caller side, if it gets an error, it logs the error message and uses the original object to compare.

``` golang
func Diff(old, new Manifest, opts ...diff.Option) (*diff.Result, error) {
	normalizedOld, err := remarshal(old.u)
	if err != nil {
                 log()
	} else {
                old.U = normalizedOld
        }
}

func remarshal(obj *unstructured.Unstructured) (*unstructured.Unstructured, error) {
```

Reply:
I got your point.
Thank you!

Reply:
I think Diff function does not have logger, so we have to pass it to Diff and some other function. Is it ok?

Reply:
Yes, I think that is fine.

Reply:
Ok, thank you!